pkg/controller/servicemesh/controlplane/reconciler.go

```diff
@@ -110,6 +110,10 @@ func (r *ControlPlaneReconciler) Reconcile() (reconcile.Result, error) {
 		"istio/charts/grafana",
 		"istio/charts/kiali",
 	}
+	if !common.IsCNIEnabled {
```

Comment:
I think a better way is to force the `spec.istio.istio_cni.enabled` flag to true/false prior to rendering the charts.  Removing this from the list of ordered components will not prevent it from being installed.

Reply:
You definitely need to set {{istio_cni.enabled}} to true, because it's also checked in some charts (even the upstream charts use the setting).